### PR TITLE
Password validations

### PR DESCRIPTION
**Pull request by jonnorstrom**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
-      messages << EXISTING_USERNAME
+
+    if password.nil?
+      messages << BLANK_REGISTER_PARAMS
+    else
+      messages << errors.messages[:password]
     end
-    return messages
+
+    User.find_by(username: username) ? messages << EXISTING_USERNAME : nil
```
Comment (DocOnDev) [inline app/models/user.rb]: would messages << EXISTING_USERNAME if User.find_by(username: username) work here? I feel like a ternary with a nil as the second half is a bit funky.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+
+    if username.length == 0
       messages << BLANK_REGISTER_PARAMS
     elsif !username.length.between?(MIN_USERNAME_LENGTH,MAX_USERNAME_LENGTH) || !username.match(VALID_USERNAME_CHARS)
       messages << INVALID_USERNAME
     end
-    if User.find_by(username: username)
-      messages << EXISTING_USERNAME
+
+    if password.nil?
```
Comment (DocOnDev) [inline app/models/user.rb]: Whereas here, couldn't we do message << password.nil? ? BLANK_REGISTER_PARMS : errors.messages[:password] and condense this to s single line? I am neutral on this, just an observation.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+      messages << BLANK_REGISTER_PARAMS
+    else
+      messages << errors.messages[:password]
     end
-    return messages
+
+    User.find_by(username: username) ? messages << EXISTING_USERNAME : nil
+
+    messages.flatten.uniq
```
Comment (DocOnDev) [inline app/models/user.rb]: smart

File: features/step_definitions/registration_steps.rb [hunk 4]
```diff
@@ -20,44 +21,43 @@
 end
 
 Then ("I do not have an account") do
-  expect(User.find_by(username: "TheEdge")).to be_falsey
+  expect(@pre_register_user_count).to eq(User.count)
```
Comment (DocOnDev) [inline features/step_definitions/registration_steps.rb]: mucho gusto. Test is much less brittle.

File: app/models/user.rb (hunk 2 above)
Comment (jonnorstrom) [reply app/models/user.rb]: Good looks. We can mess with the syntax on the next push. The 'nil' the end of the ternary works just fine - just looks funny. This bit of code could be put in to ternary as well. I believe they were written a few days apart, potentially not by the same person.

**Pull request merged**